### DISCUSSION TITLE
Understand OpenGrok release strategy.

### DISCUSSION THREAD

**Discussion by tchikh411** [oracle/opengrok#4291] [General]
We run OpenGrok for our cross-referencing needs in production environment. We have more than 500 projects totalling upto 5TB. We generally keep our production version 2 to 3 releases behind the latest releases and upgrade to the newer releases after internal tesing in test environment. 

In past few weeks, we have noticed that OpenGrok has released multiple minor releases and few major releases. Can you give us more information on the release strategy? It is also mentioned in the documentation that it is recommended to reindex from scratch on each major release change. Is this still the case? I do not see Lucene upgraded to newer version after 1.9.x

**Comment by vladak** [comment]
To be precise, these were [minor and micro/patch](https://semver.org/) releases. I don't think there will ever be major release, however I could be mistaken.

The minor version bump usually happens whenever there is **something** in the data produced by the indexer that needs to be recreated from scratch, so not necessarily tied to Lucene upgrade. Sometimes, these changes only matter if there is a particular feature enabled, however I decided not to play any games with that and simply bump the appropriate semantic version component every time this happens. Usually the release notes explain the reason for reindex from scratch.

As for why so many (minor) releases in the last period: it's simply I have some time on my hands that allow me to address some issues that have been bothering me for a long time and usually the nature of the fixes require more radical changes. Releasing frequently allows others to test the releases and if there is a problem it will make root causing easier by not having so many features packed into a release. It still holds that my intention is to produce production ready releases, whatever the semver component is. Also, by producing minor releases I keep my (and hopefully others') procedures of reindexing from scratch well oiled.

**Reply by tchikh411** [reply]
Thanks @vladak.. We should update the OpenGrok wiki page for [Release](https://github.com/oracle/opengrok/wiki/Release-process) with Semantic Versioning 2.0.0. 

I will create an issue and update the document.
